tests/test-util/test.manifest.js

```diff
@@ -0,0 +1,105 @@
+import path from 'path';
+import {assert} from 'chai';
+
+import {onlyInstancesOf, InvalidManifest} from '../../src/errors';
+import * as fs from '../../src/util/promised-fs';
+import getValidatedManifest from '../../src/util/manifest';
+import {makeSureItFails, withTempDir} from '../helpers';
+
+
+describe('util/manifest', () => {
+
+  describe('getValidatedManifest', () => {
+
+    it('returns a valid manifest', () => {
+      let expectedData = {name: 'the extension', version: '0.0.1'};
+      return withTempDir(
+        (tmpDir) =>
+          writeManifest(tmpDir.path(), expectedData)
+          .then((manifestFile) => getValidatedManifest(manifestFile))
+          .then((manifestData) => {
+            assert.deepEqual(manifestData, expectedData);
+          })
+      );
+    });
+
+    it('reports an error for a missing manifest file', () => {
+      let nonExistantFile = '/dev/null/nowhere/manifest.json';
+      return getValidatedManifest(nonExistantFile)
+        .then(makeSureItFails())
+        .catch(onlyInstancesOf(InvalidManifest, (error) => {
+          assert.match(error.message, /Could not read manifest\.json/);
+          // Make sure the filename is included in the exception message.
+          // This is actually done by default in file system error messages.
+          assert.include(error.message, nonExistantFile);
+        }));
+    });
+
+    it('reports an error for invalid manifest JSON', () => withTempDir(
+      (tmpDir) => {
+        let badManifest = 'NOT ... VALID ... JSON';
```

Comment:
nit, how about using a more realistic invalid manifest.json? something like:
```js
let badManifest = `{
  "name": "I'm an invalid JSON Manifest
  "version": "0.0.0"
}`;
```

it is not going to make a big difference for the current test case, but if we decide to enhance the json validation in a follow-up (e.g. by validating the JSON using [jsonlint](https://github.com/zaach/jsonlint) npm package as a library) it can be an useful example about how to test realistic manifest errors and be sure we catch and report them as planned.